packages/custom-types/root-state.ts

```diff
@@ -0,0 +1,58 @@
+import { MessageTypes, Wargame } from '.';
+
+export type GameInfo = {
+  imageUrlSet: boolean;
+  imageUrl: string;
+  title: string;
+  description: string;
+  showAccessCodes: boolean;
+}
+
+export type Messages = {
+  isLoading: boolean;
+  messages: never[];
+  messagePreviewId: string;
+  messagePreview: {};
+}
+
+export type UmpireMenu = {
+  selectedSchemaID: string;
+  previewSchema: {};
+}
+
+export type ExportItems = {
+  data: never[];
+  loader: boolean;
+}
+
+export interface ActivityLogsInterface {
+  activityTime: string
+  activityType: string
+  role: string
+  wargame: string
+}
+
+export type DBLoadingState = {
+  loadingMessageTypes: boolean;
+  loadingWargames: boolean;
+}
+
+export type CurrentModal = {
+  open: boolean;
+  modal: string;
+  data: null;
+}
+
+export type RootState = {
```

Comment:
Hello @lebaphi - can you explain what this new RootState is for? 

Was it an unstructured `any` in the past?

Reply:
yep, in the past we use
```
export default createStore(combineReducers({
  ....
}), applyMiddleware(...middlewares))

```

we don't define the type the state of the store, so it uses the `DefaultRootState` as default, then all state which select from store will have the `any`.
Now I create an interface for the state, I called it `RootState` with all type  of all reducers. So we can query the state that we want with its property like the image below:

<img width="629" alt="image" src="https://user-images.githubusercontent.com/26705621/158159429-1c4dacc6-e0c6-492b-85cd-dc246bfe7fda.png">